lib/wda/webdriveragent.js

```diff
@@ -156,12 +156,36 @@ class WebDriverAgent {
     }
   }
 
+
+  /**
+   * Return current running WDA's status like below after launching WDA
+   * {
+   *   "state": "success",
+   *   "os": {
+   *     "name": "iOS",
+   *     "version": "11.4",
+   *     "sdkVersion": "11.3"
+   *   },
+   *   "ios": {
+   *     "simulatorVersion": "11.4",
+   *     "ip": "172.254.99.34"
+   *   },
+   *   "build": {
+   *     "time": "Jun 24 2018 17:08:21",
+   *     "productBundleIdentifier": "com.facebook.WebDriverAgentRunner"
+   *   }
+   * }
+   *
+   * @param {string} sessionId Launch WDA and establish the session with this sessionId
+   * @return {?object} State Object
+   * @throws {Error} If there was invalid response code or body
+   */
   async launch (sessionId) {
     if (this.webDriverAgentUrl) {
       log.info(`Using provided WebdriverAgent at '${this.webDriverAgentUrl}'`);
       this.url = this.webDriverAgentUrl;
       this.setupProxies(sessionId);
-      return;
+      return this.getStatus();
```

Comment:
a note:
https://github.com/appium/appium-xcuitest-driver/pull/909/files#diff-dd1c1e9a4d2cba48fc21e5d2c50abcafR224 returns `/status` after create session.

Reply:
`return await this.getStatus();` would look better

Reply:
should we also wrap this call into try/catch?

Reply:
No, because `getStatus` already returns `null` if something happens in the `/status` command.
It's enough to return the result here